R/pkg/R/sparkR.R

```diff
@@ -155,6 +155,10 @@ sparkR.sparkContext <- function(
 
   existingPort <- Sys.getenv("EXISTING_SPARKR_BACKEND_PORT", "")
   if (existingPort != "") {
+    if (length(sparkPackages) != 0) {
```

Comment:
I looked into this and we get the unit test error because the check here isn't correct. So we get `sparkPackages`  as `""` and in R the length of empty string is not 0 but 1. 
```
> length("")
[1] 1
```
I think we should instead check the length of `packages` to be zero  ? (`packages` is a list created by splitting the input in `processSparkPackages`)